### PR TITLE
[postbuildlint] Don't read lib info twice

### PR DESCRIPTION
**Pull request by Thomas1664**
Library information was read twice from the same files: once in `check_lib_architecture` and once in `check_crt_linkage_of_libs`. This caused a performance regression. This PR moves reading the libraries to a single function (`get_lib_info`) which is only called once.

Additionally, `check_lib_architecture` performed the check `cmake_system_name == "Darwin"` but this function is only called on Windows. Therefore, this check is unnecessary and I removed the code inside it.

### CODE REVIEW

**Comment by Thomas1664** [comment]
> Additionally, check_lib_architecture performed the check cmake_system_name == "Darwin" but this function is only called on Windows. Therefore, this check is unnecessary and I removed the code inside it.

@BillyONeal I just realised that you messed this up in #834. Considering that fixing this will most likely need verification with microsoft/vcpkg and e2e tests, how to proceed with this?

**Comment by BillyONeal** [comment]
> @BillyONeal I just realised that you messed this up in #834. Considering that fixing this will most likely need verification with microsoft/vcpkg and e2e tests, how to proceed with this?

I don't think we've ever seen this break before so I'd be happy to take your fix without new tests. (I would love new tests, but I'm not enough of a macOS person to write them myself and I don't think 'restoring what should have been the status quo' qualifies as the kind of 'new feature' ask for which we require tests.)

File: src/vcpkg/postbuildlint.cpp [hunk 1]
```diff
@@ -1394,13 +1381,22 @@ namespace vcpkg
         if (windows_target)
         {
             Debug::println("Running windows targeting post-build checks");
+
+            auto release_lib_info = get_lib_info(fs, release_libs);
+            Optional<std::vector<Optional<LibInformation>>> debug_lib_info;
+
+            if (!build_info.policies.is_enabled(BuildPolicy::SKIP_ARCHITECTURE_CHECK) ||
```
Comment (BillyONeal) [inline src/vcpkg/postbuildlint.cpp:1391]: I'm not a fan of adding conditionals that need to be paired like this or the world ends but I don't have better ideas for now.

**Review by BillyONeal — APPROVED** [review]

**Comment by Thomas1664** [comment]
Do you want me to restore the lib check on OSX in another PR?

**Pull request merged**